.circleci/config.yml

```diff
@@ -28,7 +28,6 @@ commands:
             mkdir -p test-reports
             . /opt/conda/etc/profile.d/conda.sh
             conda activate esmvaltool
-            flake8 -j 4
```

Comment:
are we not running flake8 since pre-commit runs flake8-compatible tests?

Reply:
Yes: I enabled only those linting rules for `ruff` that are exactly the same as what flake8 provides (see [here](https://docs.astral.sh/ruff/faq/#how-does-ruffs-linter-compare-to-flake8) for more info). Therefore ruff replaces flake8. In the future we could consider enabling more linter rules.

Reply:
And indeed those are already run on pre-commit.ci (also mypy), so I didn't see the point in running them again on CircleCI.